packages/adapter-api/src/elements.ts

```diff
@@ -327,6 +327,7 @@ export class ObjectType extends Element {
       annotationTypes: clonedAnnotationTypes,
       annotations: clonedAnnotations,
       isSettings,
+      path: this.path,
```

Comment:
I think we should create a new path and not point on the same array
```suggestion
      path: [...this.path],
```